src/python/tests/core/bot/fuzzers/libFuzzer/engine_test.py

```diff
@@ -203,54 +211,217 @@ def mock_merge(*args, **kwargs):  # pylint: disable=unused-argument
         merge_timeout=1800.0,
         tmp_dir='/fuzz-inputs/temp-9001/merge-workdir')
 
-    self.assertDictEqual(
-        {
-            'actual_duration': 2,
-            'average_exec_per_sec': 21,
-            'bad_instrumentation': 0,
-            'corpus_crash_count': 0,
-            'corpus_size': 0,
-            'crash_count': 1,
-            'dict_used': 1,
-            'edge_coverage': 1603,
-            'edges_total': 398467,
-            'expected_duration': 1450,
-            'feature_coverage': 3572,
-            'fuzzing_time_percent': 0.13793103448275862,
-            'initial_edge_coverage': 1603,
-            'initial_feature_coverage': 3572,
-            'leak_count': 0,
-            'log_lines_from_engine': 2,
-            'log_lines_ignored': 67,
-            'log_lines_unwanted': 0,
-            'manual_dict_size': 0,
-            'max_len': 9001,
-            'merge_edge_coverage': 0,
-            'new_edges': 0,
-            'new_features': 0,
-            'new_units_added': 1,
-            'new_units_generated': 0,
-            'number_of_executed_units': 1249,
-            'oom_count': 0,
-            'peak_rss_mb': 1197,
-            'recommended_dict_size': 0,
-            'slow_unit_count': 0,
-            'slow_units_count': 0,
-            'slowest_unit_time_sec': 0,
-            'startup_crash_count': 0,
-            # TODO(ochang): Move strategy stats to common place, rather than in
-            # engine implementation.
-            'strategy_corpus_mutations_ml_rnn': 0,
-            'strategy_corpus_mutations_radamsa': 0,
-            'strategy_corpus_subset': 0,
-            'strategy_dataflow_tracing': 0,
-            'strategy_fork': 0,
-            'strategy_mutator_plugin': 0,
-            'strategy_random_max_len': 0,
-            'strategy_recommended_dict': 0,
-            'strategy_selection_method': 'default',
-            'strategy_value_profile': 0,
-            'timeout_count': 0,
-            'timeout_limit': 123,
-        },
-        result.stats)
+    self.assertDictEqual({
```

Comment:
Why so?

Reply:
Actually, the pattern is OK in this context because we are actually testing combined functionality.
But as I pointed out in [this PR](https://github.com/google/clusterfuzz/pull/583), when we test a new stat we assert that stats dictionary is equal to a hardcoded one containing the stat we expect rather than testing for the presence of the stat.
This technical debt requires copying tons of code and updating tons of tests that have nothing to do with the feature being added. Still hoping it gets removed where harmful in this refactor.

Reply:
Yeah, definitely agree with this issue in the existing tests.